### PR TITLE
Add Obsidian.app

### PR DESCRIPTION
**Pull request by apetresc**
**Important:** *Do not tick a checkbox if you haven’t performed its action.* Honesty is indispensable for a smooth review process.

After making all changes to a cask, verify:

- [x] `brew cask audit --download {{cask_file}}` is error-free.
- [x] `brew cask style --fix {{cask_file}}` reports no offenses.
- [x] There are no [open pull requests](https://github.com/Homebrew/homebrew-cask/pulls) for the same update.
- [x] The submission is for [a stable version](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#stable-versions) or [documented exception](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#but-there-is-no-stable-version).

Additionally, **if adding a new cask**:

- [x] Named the cask according to the [token reference](https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/token_reference.md).
- [x] `brew cask install {{cask_file}}` worked successfully.
- [x] `brew cask uninstall {{cask_file}}` worked successfully.
- [x] Checked the cask was not [already refused](https://github.com/Homebrew/homebrew-cask/search?q=is%3Aclosed&type=Issues).
- [x] Checked the cask is submitted to [the correct repo](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#finding-a-home-for-your-cask).

### CODE REVIEW

File: Casks/obsidian.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
+cask 'obsidian' do
+  version '0.6.4'
+  sha256 '1da5fffd869201ba044137ebfaf0f0659982a653f3f5b75d6634c410f8eb6b8c'
+
+  # github.com was verified as official when first introduced to the cask
+  url 'https://github.com/obsidianmd/obsidian-releases/releases/download/v0.6.4/Obsidian-0.6.4.dmg'
```
Comment (miccal) [inline Casks/obsidian.rb:6]: Please change this to:
```
url "https://github.com/obsidianmd/obsidian-releases/releases/download/v#{version}/Obsidian-#{version}.dmg"
```

File: Casks/obsidian.rb [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  appcast 'https://github.com/obsidianmd/obsidian-releases/releases.atom'
+  name '0.6.4'
```
Comment (miccal) [inline Casks/obsidian.rb:8]: Please change this to:
```
name 'Obsidian'
```

File: Casks/obsidian.rb [hunk 3]
```diff
@@ -0,0 +1,18 @@
+cask 'obsidian' do
+  version '0.6.4'
+  sha256 '1da5fffd869201ba044137ebfaf0f0659982a653f3f5b75d6634c410f8eb6b8c'
+
+  # github.com was verified as official when first introduced to the cask
```
Comment (miccal) [inline Casks/obsidian.rb:5]: Please change this to:
```
# github.com/obsidianmd/ was verified as official when first introduced to the cask
```

**Comment by ran-dall** [comment]
@miccal On point, mate! I actually was going to push these fixes, but I ended up getting a phone call. Good looking out. 👍

**Comment by ran-dall** [comment]
Due to the positive reactions of some fellow HBC members and the engagement on Product Hunt, I'll merge this.

**Comment by chrisfinazzo** [comment]
> Due to the positive reactions of some fellow HBC members and the engagement on Product Hunt, I'll merge this.

Glad to see this merged.

I had a gist with this ready to go yesterday, but talking to them on Twitter, I think they originally didn't ship a zip - it was a `.asar.gz`, whatever that is - which wasn't going to work. I _think_ the DMG was there all along.

**Pull request merged**